test/functional/logforwarding/message_format_test.go

```diff
@@ -0,0 +1,61 @@
+package logforwarding
+
+import (
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	logging "github.com/openshift/cluster-logging-operator/pkg/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/test/functional"
+	"github.com/openshift/cluster-logging-operator/test/helpers"
+)
+
+var _ = Describe("[LogForwarding] Functional tests for message format", func() {
+
+	var (
+		framework *functional.FluentdFunctionalFramework
+	)
+
+	BeforeEach(func() {
+		framework = functional.NewFluentdFunctionalFramework()
+		functional.NewClusterLogForwarderBuilder(framework.Forwarder).
+			FromInput(logging.InputNameApplication).
+			ToFluentForwardOutput()
+		Expect(framework.Deploy()).To(BeNil())
+	})
+	AfterEach(func() {
+		framework.Cleanup()
+	})
+
+	It("should parse application log format correctly", func() {
+
+		// Define a template for test format (used for input, and expected output)
+		var testFormatTemplate = helpers.Log{
+			Timestamp: "2020-11-04T18:13:59.061892+00:00",
+			Message:   "Pre-Functional test message",
+			IndexName: "app-write",
+			Level:     "unknown",
+			MessageID: "*",
+			Docker:    &helpers.Docker{
+				ContainerID: "*"},
+			Kubernetes: &helpers.K8s{
+				ContainerName: "*",
+				PodName: "*",
+				NamespaceName:"*",
+				NamespaceID: "*",
+				OrphanedNamespace: "*"},
+		}
+
+		// Write log line as input to fluentd
+		applicationLogLine := fmt.Sprintf("%s stdout F %s $n", testFormatTemplate.Timestamp, testFormatTemplate.Message)
+		Expect(framework.WritesMessageToApplicationLogs(applicationLogLine,1)).To(BeNil())
+		// Read line from Log Forward output
+		raw, err := framework.ReadApplicationLogsFrom("fluentforward")
```

Comment:
While testing my functional benchmarker I had issues with this method returning correctly because of the timestamp was in the past.  I would verify this actually works for  you in that you may need to pass a "Now" timestamp in order for fluent to not buffer the logs elsewhere

Reply:
@jcantrill In this test since we are always writing a single log line and reading back a single line there is no issue. This should be extended for more advanced tests like the benchmarking but not as part of this PR. I ran this large amount of times and this was always working for me.